gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/MRCompactorAvroKeyDedupJobRunner.java

```diff
@@ -161,17 +167,60 @@ private boolean useAllAttributesForCompaction() {
         ConfigurationKeys.DEFAULT_COMPACTION_USE_ALL_ATTRIBUTES);
   }
 
-  private Schema getAllFieldsExceptMap(Schema schema) {
+  /**
+   * Remove map, array, enum fields, as well as union fields that contain map, array or enum,
+   * from an Avro schema.
+   */
+  private static Optional<Schema> removeUncomparableFields(Schema schema) {
```

Comment:
This seems like a useful utility method. Can we move it to a public utility class?